src/watch/fileWatchers.ts

```diff
@@ -28,17 +30,19 @@ export function addTask(
 		}
 	}
 
-	group.get(id).tasks.add(task);
+	if (isTransformDependency) watcher.transformDependencyTasks.add(task);
+	else watcher.tasks.add(task);
```

Comment:
Wouldn't it be nicer to not reach into the internals of the watcher but add a method to the watcher `addTask(task, isTransformDependency)` and let the watcher decide how the type of dependency is handled itself and make `tasks`private?

Reply:
Also it looks like if a dependency is both an transform and a normal dependency, it will be watched twice.